tests/openassetio-core/trait/TraitBaseTest.cpp

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#include <openassetio/trait/TraitBase.hpp>
+#include <openassetio/typedefs.hpp>
+
+#include <catch2/catch.hpp>
+#include "openassetio/TraitsData.hpp"
+
+namespace {
+
+using namespace openassetio;
+
+// TraitBase can't be tested directly, so we derive a test trait.
+struct TestTrait : trait::TraitBase<TestTrait> {
+ private:
+  static inline const trait::property::Key kSomeProperty{"some property"};
+
+ public:
+  static inline const trait::TraitId kId{"test"};
+
+  using trait::TraitBase<TestTrait>::TraitBase;
+
+  [[nodiscard]] trait::TraitPropertyStatus getSomeProperty(Str* out) const {
+    return getTraitProperty(out, kId, kSomeProperty);
+  }
+
+  void setSomeProperty(const Str& value) { data()->setTraitProperty(kId, kSomeProperty, value); }
+
+  // Helpers for tests
+
+  [[nodiscard]] TraitsDataPtr& dataPtr() { return data(); }
+  [[nodiscard]] const TraitsDataPtr& dataPtr() const { return data(); }
+};
+
+SCENARIO("Trait data method") {
+  WHEN("constructed with data") {
+    TraitsDataPtr data = std::make_shared<TraitsData>();
+    TestTrait trait(data);
+
+    THEN("the supplied data is exposed via protected method") { CHECK(trait.dataPtr() == data); }
+
+    THEN("the supplied data is exposed via const protected method") {
+      const TestTrait& constTrait = trait;
+      CHECK(constTrait.dataPtr() == data);
+    }
+  }
+}
+
+SCENARIO("Trait isValid") {
```

Comment:
[consider][minor] I think these scenarios could be a bit more descriptive in the BDD language, e.g.

```c++
SCENARIO("Checking a trait is valid")
  GIVEN("Some known data")
    AND_GIVEN("A non-const trait instance constructed from known data")
      WHEN("the trait queries its underlying data")
        THEN("the underlying data points to the known data")
    AND_GIVEN("A const trait instance constructed from known data")
     ...
```

... Minor though, it's still pretty obvious what these tests are doing.

Reply:
Hows [this](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/373/commits/8812717dc45b4708dc6cadfbbfc4af3ae6e9851d)?

Reply:
Nice. Though see other upcoming comments.